Release 5.39.0 Has IPhysicsEnginePLugin Errors

The 5.39.0 Release introduces new bugs in the Babylon.d.ts

When type to build my lib against the new declaration file using TSC 4.6.2. It gets plenty errors now. It did not get those error before.

```
[17:25:24] Using gulpfile ~/Documents/Babylon/BabylonToolkit-Share/Manager/gulpfile.js
[17:25:24] Starting 'default'...
[17:25:24] Starting 'compile'...
types/babylon.d.ts(80474,9): error TS2416: Property 'executeStep' in type 'AmmoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(delta: number, impostors: PhysicsImpostor[]) => void' is not assignable to type '{ (delta: number, impostors: PhysicsImpostor[]): void; (delta: number): void; }'.
types/babylon.d.ts(80497,9): error TS2416: Property 'applyImpulse' in type 'AmmoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, force: Vector3, contactPoint: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, force: Vector3, contactPoint: Vector3): void; (body: PhysicsBody, location: Vector3, impulse: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is missing the following properties from type 'PhysicsImpostor': object, type, _options, _physicsEngine, and 66 more.
types/babylon.d.ts(80577,9): error TS2416: Property 'setLinearVelocity' in type 'AmmoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, velocity: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, velocity: Vector3): void; (body: PhysicsBody, linVel: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
types/babylon.d.ts(80583,9): error TS2416: Property 'setAngularVelocity' in type 'AmmoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, velocity: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, velocity: Vector3): void; (body: PhysicsBody, angVel: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
types/babylon.d.ts(80786,9): error TS2416: Property 'executeStep' in type 'CannonJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(delta: number, impostors: PhysicsImpostor[]) => void' is not assignable to type '{ (delta: number, impostors: PhysicsImpostor[]): void; (delta: number): void; }'.
types/babylon.d.ts(80788,9): error TS2416: Property 'applyImpulse' in type 'CannonJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, force: Vector3, contactPoint: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, force: Vector3, contactPoint: Vector3): void; (body: PhysicsBody, location: Vector3, impulse: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
types/babylon.d.ts(80808,9): error TS2416: Property 'setLinearVelocity' in type 'CannonJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, velocity: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, velocity: Vector3): void; (body: PhysicsBody, linVel: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
types/babylon.d.ts(80809,9): error TS2416: Property 'setAngularVelocity' in type 'CannonJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, velocity: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, velocity: Vector3): void; (body: PhysicsBody, angVel: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
types/babylon.d.ts(80865,9): error TS2416: Property 'executeStep' in type 'OimoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(delta: number, impostors: PhysicsImpostor[]) => void' is not assignable to type '{ (delta: number, impostors: PhysicsImpostor[]): void; (delta: number): void; }'.
types/babylon.d.ts(80866,9): error TS2416: Property 'applyImpulse' in type 'OimoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, force: Vector3, contactPoint: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, force: Vector3, contactPoint: Vector3): void; (body: PhysicsBody, location: Vector3, impulse: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
types/babylon.d.ts(80876,9): error TS2416: Property 'setLinearVelocity' in type 'OimoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, velocity: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, velocity: Vector3): void; (body: PhysicsBody, linVel: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
types/babylon.d.ts(80877,9): error TS2416: Property 'setAngularVelocity' in type 'OimoJSPlugin' is not assignable to the same property in base type 'IPhysicsEnginePlugin'.
  Type '(impostor: PhysicsImpostor, velocity: Vector3) => void' is not assignable to type '{ (impostor: PhysicsImpostor, velocity: Vector3): void; (body: PhysicsBody, angVel: Vector3): void; }'.
    Types of parameters 'impostor' and 'body' are incompatible.
      Type 'PhysicsBody' is not assignable to type 'PhysicsImpostor'.
TypeScript: 12 semantic errors
TypeScript: emit succeeded (with errors)
[17:25:38] Finished 'compile' after 14 s
[17:25:38] Starting '<anonymous>'...
[17:25:41] Finished '<anonymous>' after 2.77 s
[17:25:41] Finished 'default' after 17 s
```

--- a/packages/dev/core/src/Debug/physicsViewer.ts
+++ b/packages/dev/core/src/Debug/physicsViewer.ts
@@ -10,7 +10,7 @@ import type { Material } from "../Materials/material";
 import { EngineStore } from "../Engines/engineStore";
 import { StandardMaterial } from "../Materials/standardMaterial";
 import type { IPhysicsEnginePlugin as IPhysicsEnginePluginV1 } from "../Physics/v1/IPhysicsEnginePlugin";
-import type { IPhysicsEnginePlugin as IPhysicsEnginePluginV2 } from "../Physics/v2/IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2 } from "../Physics/v2/IPhysicsEnginePlugin";
 import { PhysicsImpostor } from "../Physics/v1/physicsImpostor";
 import { UtilityLayerRenderer } from "../Rendering/utilityLayerRenderer";
 import { CreateCylinder } from "../Meshes/Builders/cylinderBuilder";

--- a/packages/dev/core/src/Physics/IPhysicsEngine.ts
+++ b/packages/dev/core/src/Physics/IPhysicsEngine.ts
@@ -1,7 +1,7 @@
 import type { Vector3 } from "../Maths/math.vector";
 import type { PhysicsRaycastResult } from "./physicsRaycastResult";
 import type { IPhysicsEnginePlugin as IPhysicsEnginePluginV1 } from "./v1/IPhysicsEnginePlugin";
-import type { IPhysicsEnginePlugin as IPhysicsEnginePluginV2 } from "./v2/IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2 } from "./v2/IPhysicsEnginePlugin";
 
 /**
  * Interface used to define a physics engine

--- a/packages/dev/core/src/Physics/physicsEngineComponent.ts
+++ b/packages/dev/core/src/Physics/physicsEngineComponent.ts
@@ -8,7 +8,7 @@ import { SceneComponentConstants } from "../sceneComponent";
 import { Scene } from "../scene";
 import type { IPhysicsEngine } from "./IPhysicsEngine";
 import type { IPhysicsEnginePlugin as IPhysicsEnginePluginV1 } from "./v1/IPhysicsEnginePlugin";
-import type { IPhysicsEnginePlugin as IPhysicsEnginePluginV2 } from "./v2/IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2 } from "./v2/IPhysicsEnginePlugin";
 import { PhysicsEngine as PhysicsEngineV1 } from "./v1/physicsEngine";
 import { PhysicsEngine as PhysicsEngineV2 } from "./v2/physicsEngine";
 

--- a/packages/dev/core/src/Physics/v2/IPhysicsEnginePlugin.ts
+++ b/packages/dev/core/src/Physics/v2/IPhysicsEnginePlugin.ts
@@ -97,7 +97,7 @@ export interface MassProperties {
 }
 
 /** @internal */
-export interface IPhysicsEnginePlugin {
+export interface IPhysicsEnginePluginV2 {
     /**
      *
      */

--- a/packages/dev/core/src/Physics/v2/physicsBody.ts
+++ b/packages/dev/core/src/Physics/v2/physicsBody.ts
@@ -1,4 +1,4 @@
-import type { IPhysicsEnginePlugin, MassProperties } from "./IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2, MassProperties } from "./IPhysicsEnginePlugin";
 import type { PhysicsShape } from "./physicsShape";
 import type { Vector3 } from "../../Maths/math.vector";
 import type { Scene } from "../../scene";
@@ -11,7 +11,7 @@ export class PhysicsBody {
     /** @internal */
     public _pluginData: any = undefined;
 
-    private _physicsPlugin: IPhysicsEnginePlugin;
+    private _physicsPlugin: IPhysicsEnginePluginV2;
 
     /**
      *
@@ -34,7 +34,7 @@ export class PhysicsBody {
             throw new Error("No Physics Plugin available.");
         }
 
-        this._physicsPlugin = physicsPlugin as IPhysicsEnginePlugin;
+        this._physicsPlugin = physicsPlugin as IPhysicsEnginePluginV2;
         this._physicsPlugin.initBody(this);
     }
     /**

--- a/packages/dev/core/src/Physics/v2/physicsConstraint.ts
+++ b/packages/dev/core/src/Physics/v2/physicsConstraint.ts
@@ -1,6 +1,6 @@
 import type { Scene } from "../../scene";
 import type { Vector3 } from "../../Maths/math.vector";
-import type { IPhysicsEnginePlugin, ConstraintAxis, PhysicsConstraintParameters, ConstraintAxisLimitMode, ConstraintMotorType } from "./IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2, ConstraintAxis, PhysicsConstraintParameters, ConstraintAxisLimitMode, ConstraintMotorType } from "./IPhysicsEnginePlugin";
 import { ConstraintType } from "./IPhysicsEnginePlugin";
 import type { PhysicsBody } from "./physicsBody";
 
@@ -16,7 +16,7 @@ export class PhysicsConstraint {
      *
      */
     public _pluginData: any = undefined;
-    protected _physicsPlugin: IPhysicsEnginePlugin;
+    protected _physicsPlugin: IPhysicsEnginePluginV2;
 
     /**
      *
@@ -37,7 +37,7 @@ export class PhysicsConstraint {
             throw new Error("No Physics Plugin available.");
         }
 
-        this._physicsPlugin = physicsPlugin as IPhysicsEnginePlugin;
+        this._physicsPlugin = physicsPlugin as IPhysicsEnginePluginV2;
         this._physicsPlugin.initConstraint(this, type, options);
     }
 

--- a/packages/dev/core/src/Physics/v2/physicsEngine.ts
+++ b/packages/dev/core/src/Physics/v2/physicsEngine.ts
@@ -1,7 +1,7 @@
 import type { Nullable } from "../../types";
 import { Vector3 } from "../../Maths/math.vector";
 import type { IPhysicsEngine } from "../IPhysicsEngine";
-import type { IPhysicsEnginePlugin } from "./IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2 } from "./IPhysicsEnginePlugin";
 import { PhysicsRaycastResult } from "../physicsRaycastResult";
 import { _WarnImport } from "../../Misc/devTools";
 
@@ -34,7 +34,7 @@ export class PhysicsEngine implements IPhysicsEngine {
      * Factory used to create the default physics plugin.
      * @returns The default physics plugin
      */
-    public static DefaultPluginFactory(): IPhysicsEnginePlugin {
+    public static DefaultPluginFactory(): IPhysicsEnginePluginV2 {
         throw _WarnImport("");
     }
 
@@ -43,7 +43,7 @@ export class PhysicsEngine implements IPhysicsEngine {
      * @param gravity defines the gravity vector used by the simulation
      * @param _physicsPlugin defines the plugin to use (CannonJS by default)
      */
-    constructor(gravity: Nullable<Vector3>, private _physicsPlugin: IPhysicsEnginePlugin = PhysicsEngine.DefaultPluginFactory()) {
+    constructor(gravity: Nullable<Vector3>, private _physicsPlugin: IPhysicsEnginePluginV2 = PhysicsEngine.DefaultPluginFactory()) {
         gravity = gravity || new Vector3(0, -9.807, 0);
         this.setGravity(gravity);
         this.setTimeStep();
@@ -141,7 +141,7 @@ export class PhysicsEngine implements IPhysicsEngine {
      * Gets the current plugin used to run the simulation
      * @returns current plugin
      */
-    public getPhysicsPlugin(): IPhysicsEnginePlugin {
+    public getPhysicsPlugin(): IPhysicsEnginePluginV2 {
         return this._physicsPlugin;
     }
 

--- a/packages/dev/core/src/Physics/v2/physicsMaterial.ts
+++ b/packages/dev/core/src/Physics/v2/physicsMaterial.ts
@@ -1,5 +1,5 @@
 import type { Scene } from "../../scene";
-import type { IPhysicsEnginePlugin } from "./IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2 } from "./IPhysicsEnginePlugin";
 
 /**
  *
@@ -12,7 +12,7 @@ export class PhysicsMaterial {
      */
     public _pluginData: any = undefined;
 
-    protected _physicsPlugin: IPhysicsEnginePlugin;
+    protected _physicsPlugin: IPhysicsEnginePluginV2;
 
     /**
      *
@@ -33,7 +33,7 @@ export class PhysicsMaterial {
             throw new Error("No Physics Plugin available.");
         }
 
-        this._physicsPlugin = physicsPlugin as IPhysicsEnginePlugin;
+        this._physicsPlugin = physicsPlugin as IPhysicsEnginePluginV2;
         this._physicsPlugin.initMaterial(this);
     }
     /**

--- a/packages/dev/core/src/Physics/v2/physicsShape.ts
+++ b/packages/dev/core/src/Physics/v2/physicsShape.ts
@@ -1,7 +1,7 @@
 import type { TransformNode } from "../../Meshes/transformNode";
 import type { BoundingBox } from "../../Culling/boundingBox";
 import { ShapeType } from "./IPhysicsEnginePlugin";
-import type { IPhysicsEnginePlugin, PhysicsShapeParameters } from "./IPhysicsEnginePlugin";
+import type { IPhysicsEnginePluginV2, PhysicsShapeParameters } from "./IPhysicsEnginePlugin";
 import type { PhysicsMaterial } from "./physicsMaterial";
 import type { Vector3 } from "../../Maths/math.vector";
 import type { Quaternion } from "../../Maths/math.vector";
@@ -16,7 +16,7 @@ export class PhysicsShape {
     /** @internal */
     public _pluginData: any = undefined;
 
-    private _physicsPlugin: IPhysicsEnginePlugin;
+    private _physicsPlugin: IPhysicsEnginePluginV2;
 
     private _type: ShapeType;
 
@@ -45,7 +45,7 @@ export class PhysicsShape {
             throw new Error("No Physics Plugin available.");
         }
 
-        this._physicsPlugin = physicsPlugin as IPhysicsEnginePlugin;
+        this._physicsPlugin = physicsPlugin as IPhysicsEnginePluginV2;
         this._physicsPlugin.initShape(this, type, options);
     }
 

